### PR TITLE
feat: workspace nlp rfq

### PR DESCRIPTION
**Pull request by SHolleworth**
Feature for raising an RFQ from the Workspace home input bar.

image

After raising the rfq the user can open the credit rfqs view

image

### CODE REVIEW

File: src/workspace/src/services/connection.ts
Comment (SHolleworth) [inline src/workspace/src/services/connection.ts:1]: I had to copy a few pieces of our client services over to workspace

File: src/workspace/src/services/creditInstruments.ts
Comment (SHolleworth) [inline src/workspace/src/services/creditInstruments.ts:1]: As well as write some of my own modified versions of services

File: src/workspace/src/home/index.ts [hunk 1]
```diff
@@ -80,55 +82,90 @@ export async function registerHome(): Promise<HomeRegistration> {
...
+            direction,
+            notional,
+            dealtCurrency,
+          })
+        }
+
+        break
+      }
+
+      case "rfq-execution": {
```
Comment (SHolleworth) [inline src/workspace/src/home/index.ts:129]: The new case

File: src/workspace/src/services/connection.ts
Comment (algreasley) [reply src/workspace/src/services/connection.ts:1]: I'd like to converge client/workspace if at all possible .. just something to keep in mind, not an immediate goal

**Review by TobyLi96 — APPROVED** [review]

File: src/workspace/src/services/connection.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
+import {
```
[the remark this answers is not in the document]
Comment (SHolleworth) [reply src/workspace/src/services/connection.ts:1]: Good catch, I think we should actually be using initConnection so I switched that out. I've gotten rid of withIsStaleData

File: src/workspace/src/home/nlpProvider.ts [hunk 3]
```diff
@@ -5,24 +5,40 @@ import {
...
   CurrencyPair,
   Direction,
+  InstrumentBody,
+  NackCreateRfqResponse,
+  RfqBody,
   Trade,
   TradeStatus,
 } from "@/generated/TradingGateway"
+import { creditInstruments$ } from "@/services/creditInstruments"
+import { createdRfqInstrumentReponse$ as createdRfqWithInstrumentWithReponse$ } from "@/services/creditRfqRequests"
```
Comment (TobyLi96) [inline src/workspace/src/home/nlpProvider.ts:31]: minor: typo

**Review by TobyLi96 — APPROVED** [review]

**Review by TobyLi96 — APPROVED** [review]

**Pull request merged**